clang/include/clang/Basic/DiagnosticSemaKinds.td

```diff
@@ -6171,6 +6171,9 @@ def warn_signed_bitfield_enum_conversion : Warning<
   "signed bit-field %0 needs an extra bit to represent the largest positive "
   "enumerators of %1">,
   InGroup<BitFieldEnumConversion>, DefaultIgnore;
+def warn_bitfield_too_small_for_integral_type : Warning<
+  "bit-field %0 (%1 bits) is not wide enough to store type of %2 ">,
+  InGroup<Conversion>, DefaultIgnore;
```

Comment:
Thanks @erichkeane, understood. I'm thinking this warning should be disabled by default and then explicitly enabled only if the user wants to enable it. Do you agree, or have different ideas about the use cases?

Reply:
We typically do not add diagnostics that are opt-in because we've had evidence in the past that they are not enabled enough to be worth the maintenance costs. So the goal is to have very, very few false positives -- for things that are more prone to false positives, we usually put them in clang-tidy instead. However, this one is a bit of an outlier because many diagnostics under `-Wconversion` are ignored by default. So, I think this one may actually be fine to be off-by-default, but I do expect it to be quite chatty.

Have you tried seeing how many diagnostics are generated when building Clang itself? It would be interesting to see what kind of results that gives you because we use bit-fields relatively frequently.

Btw, I think the GCC phrasing for the diagnostic is an improvement over the current wording. (You don't store a type, so that's a bit awkward.)

Reply:
Thanks @AaronBallman. I've tried the change I made on our internal code base, and see quite a few interesting findings we had not seen before. I will try the change on llvm/clang, and a few open source projects and report back. I'll also consider the gcc phrasing as a suggested improvement for the message. Best!

Reply:
Carrying forward the requested task to get compiler results for llvm/clang, and resolving this thread.